perllib/Open311/GetServiceRequestUpdates.pm

```diff
@@ -262,17 +262,22 @@ sub comment_text_for_request {
     my $ext_code_changed = $ext_code ne $old_ext_code;
     my $template;
     if ($state_changed || $ext_code_changed) {
+        my $order;
         my $state_params = {
             'me.state' => $state
         };
         if ($ext_code) {
             $state_params->{'me.external_status_code'} = $ext_code;
+            # make sure that empty string comes last. Can't use DESC as that orders
+            # empty string before codes starting with 0 and explicit ASC will order
```

Comment:
This sort works fine, but I don't understand the comment - DESC sorts "0" before "" as far as I can see?

```
bci=> select state,external_status_code,external_status_code is null from response_templates
where auto_response order by external_status_code desc nulls last;
...
                  | 0015                 | f
                  | 0010                 | f
                  | 0010                 | f
investigating     |                      | f
duplicate         |                      | f
action scheduled  |                      | f
```

Reply:
weird, I definitely had issues with this when I was testing. I will see if I can work out what's going on.

Reply:
hmmm, I have no idea why it was failing like that for me :|